perceval/client.py

```diff
@@ -197,14 +197,27 @@ def sleep_for_rate_limit(self):
            raises a RateLimitError exception if sleep_for_rate flag is disabled.
         """
         if self.rate_limit is not None and self.rate_limit <= self.min_rate_to_sleep:
-            seconds_to_reset = self.rate_limit_reset_ts - int(time.time()) + 1
+            seconds_to_reset = self.__calculate_seconds_to_reset()
             cause = "Rate limit exhausted."
             if self.sleep_for_rate:
                 logger.info("%s Waiting %i secs for rate limit reset.", cause, seconds_to_reset)
                 time.sleep(seconds_to_reset)
             else:
                 raise RateLimitError(cause=cause, seconds_to_reset=seconds_to_reset)
 
+    def __calculate_seconds_to_reset(self):
```

Comment:
I suggest to define this method as abstract that should be overwritten by each specific `RateLimitHandler`. Therefore, Meetup and GitHub will have their way to manage the time to reset. The method would be renamed to `calculate_time_to_reset` to make it more generic.